vignettes/transition_disruption_metric.Rmd

```diff
@@ -0,0 +1,552 @@
+            2, 2025L,               "S(t)",    "solid",
+            3, 2030L,               "S(t)",    "solid",
+            4, 2035L,               "S(t)",    "solid",
+            5, 2040L,               "S(t)",    "solid",
+            6, 2045L,               "S(t)",    "solid",
+            7, 2050L,               "S(t)",    "solid",
+          0.5, 2020L,               "P(t)",    "solid",
+            0.6, 2025L,               "P(t)",    "solid",
+            0.6, 2025L, "Required build-out",   "dashed",
+            3, 2030L, "Required build-out",   "dashed"
+  )
+
+data %>%
+  ggplot(aes(x = year, y = production, group = type)) +
+  geom_line(aes(color = type, linetype = line_type)) +
+  scale_color_manual(values = c("#9999CC", "#0B7A75", "#CC6666")) +
+  guides(linetype = "none") +
+  scale_linetype_manual(values = c("dashed", "solid")) + 
+  guides(fill = "none") +
+  geom_point(aes(shape = type)) +
+  scale_shape_manual(values = c("triangle", "triangle", "circle")) +
+  labs(title = "Production by Year", x = "Year", y = "Production") + 
+  theme(
+    axis.text.y = element_blank()
+  ) + 
+  scale_x_continuous(
+    breaks = c(2020, 2025, 2030, 2035, 2040, 2045, 2050),
+    labels = c("t", "t+t1", "t+t2", "", "", "", "")
+    )
+```
+
+- Case 2: $\frac{m_P}{m_S} = 1$: The portfolio and scenario have the exact same value at $t+t_1$, and thus the portfolio will need to increaes it's production at exactly the same rate as the scenario (difficult to see on the graph, since "Required build-out" is on the same line as S(t)).
+
+```{r, echo = FALSE}
+
+data <- tibble::tribble(
+  ~production, ~year,                ~type, ~line_type,
+            1, 2020L,               "S(t)",    "solid",
+            2, 2025L,               "S(t)",    "solid",
+            3, 2030L,               "S(t)",    "solid",
+            4, 2035L,               "S(t)",    "solid",
+            5, 2040L,               "S(t)",    "solid",
+            6, 2045L,               "S(t)",    "solid",
+            7, 2050L,               "S(t)",    "solid",
+          0.5, 2020L,               "P(t)",    "solid",
+            2, 2025L,               "P(t)",    "solid",
+            2, 2025L, "Required build-out",   "dashed",
+            3, 2030L, "Required build-out",   "dashed"
+  )
+
+data %>%
+  ggplot(aes(x = year, y = production, group = type)) +
+  geom_line(aes(color = type, linetype = line_type)) +
+  scale_color_manual(values = c("#9999CC", "#0B7A75", "#CC6666")) +
+  guides(linetype = "none") +
+  scale_linetype_manual(values = c("dashed", "solid")) + 
+  guides(fill = "none") +
+  geom_point(aes(shape = type)) +
+  scale_shape_manual(values = c("triangle", "triangle", "circle")) +
+  labs(title = "Production by Year", x = "Year", y = "Production") + 
+  theme(
+    axis.text.y = element_blank()
+  ) + 
+  scale_x_continuous(
+    breaks = c(2020, 2025, 2030, 2035, 2040, 2045, 2050),
+    labels = c("t", "t+t1", "t+t2", "", "", "", "")
+    )
```

Comment:
This seems to be repeated. Maybe DRY in a lil helper so the focus stays in the message and not in the long code.

Reply:
Oh, I see -- the user doesn't see the output because you use `echo = FALSE`. Then:

- [ ] You may use `echo = FALSE` only once at the top.
- [ ] You may still extract a helper to yourself.
- [ ] You may want to share rendered output so I can see what the reader will see. Sure I can checkout locally, but if you make my life easy I'll like you more.